File: examples/python/documentation/remote_backend.py
# DOCS_BLOCK_START:remote-backend-define
from constructs import Construct
from cdktf import App, CloudBackend, NamedCloudWorkspace, TerraformStack, TerraformOutput
# DOCS_BLOCK_END:remote-backend-define

class Stack(TerraformStack):
    def __init__(self, scope: Construct, id: str):
        super().__init__(scope, id)

# DOCS_BLOCK_START:remote-backend-migrate
stack = Stack(App(), "hi-terraform")
# DOCS_BLOCK_END:remote-backend-migrate

# DOCS_BLOCK_START:remote-backend-define
class RemoteBackendStack(TerraformStack):
    def __init__(self, scope: Construct, id: str):
        super().__init__(scope, id)
        
        # DOCS_BLOCK_END:remote-backend-define
        
        # DOCS_BLOCK_START:remote-backend-define,remote-backend-migrate
        CloudBackend(self,
            hostname = "app.terraform.io",
            organization = "company",
            workspaces = NamedCloudWorkspace("my-app-prod")
        )
        # DOCS_BLOCK_END:remote-backend-define,remote-backend-migrate
        
        # DOCS_BLOCK_START:remote-backend-define
        
        TerraformOutput(self, "dns-server",
            value = "hello-world"
        )

# DOCS_BLOCK_END:remote-backend-define
'''
DOCS_BLOCK_START:remote-backend-define
app = App()
RemoteBackendStack(app, "hello-terraform")
app.synth()
DOCS_BLOCK_END:remote-backend-define
'''

# DOCS_BLOCK_START:remote-backend-escape-hatches
stack.add_override("terraform.backend",{
    "atlas": {
        "name": "example_corp/networking-prod",
        "address": "https://app.terraform.io"
    }
})
# DOCS_BLOCK_END:remote-backend-escape-hatches

File: examples/go/documentation/remote-backend.go
// Copyright (c) HashiCorp, Inc.
// SPDX-License-Identifier: MPL-2.0

package main

// DOCS_BLOCK_START:remote-backend-define
import (
	"github.com/aws/constructs-go/constructs/v10"
	"github.com/aws/jsii-runtime-go"
	"github.com/hashicorp/terraform-cdk-go/cdktf"
)

func NewCloudBackendStack(scope constructs.Construct, name string) cdktf.TerraformStack {
	stack := cdktf.NewTerraformStack(scope, &name)

	cdktf.NewCloudBackend(stack, &cdktf.CloudBackendConfig{
		Hostname:     jsii.String("app.terraform.io"),
		Organization: jsii.String("company"),
		Workspaces:   cdktf.NewNamedCloudWorkspace(jsii.String("my-app-prod")),
	})

	cdktf.NewTerraformOutput(stack, jsii.String("dns-server"), &cdktf.TerraformOutputConfig{
		Value: "hello-world",
	})

	return stack
}

// DOCS_BLOCK_END:remote-backend-define

// DOCS_BLOCK_START:remote-backend-migrate

func NewLocalBackendStack(scope constructs.Construct, name string) cdktf.TerraformStack {
	stack := cdktf.NewTerraformStack(scope, &name)

	cdktf.NewTerraformOutput(stack, jsii.String("dns-server"), &cdktf.TerraformOutputConfig{
		Value: "local",
	})

	return stack
}

// DOCS_BLOCK_END:remote-backend-migrate

/*
We fake the methods name to be "main"
DOCS_BLOCK_START:remote-backend-migrate
func main() {
DOCS_BLOCK_END:remote-backend-migrate
*/
func SynthLocalBackend() {
	// DOCS_BLOCK_START:remote-backend-migrate
	app := cdktf.NewApp(nil)

	stack := NewLocalBackendStack(app, "hello-terraform")
	cdktf.NewCloudBackend(stack, &cdktf.CloudBackendConfig{
		Hostname:     jsii.String("app.terraform.io"),
		Organization: jsii.String("company"),
		Workspaces:   cdktf.NewNamedCloudWorkspace(jsii.String("my-app-prod")),
	})

	// DOCS_BLOCK_END:remote-backend-migrate
	// DOCS_BLOCK_START:remote-backend-escape-hatches
	stack.AddOverride(jsii.String("terraform.backend"), &map[string]map[string]string{
		"atlas": {
			"name":    "example_corp/networking-prod",
			"address": "https://app.terraform.io",
		},
	})
	// DOCS_BLOCK_END:remote-backend-escape-hatches
	// DOCS_BLOCK_START:remote-backend-migrate
	app.Synth()
}

// DOCS_BLOCK_END:remote-backend-migrate
